src/linear_algebra/dual.lean

```diff
@@ -886,6 +897,37 @@ begin
   exact finrank_range_add_finrank_ker f,
 end
 
+open linear_map
+/-- `T` is surjective if and only if `T.dual_map` is injective -/
+@[simp] lemma dual_map_injective_iff
+  (T : V₁ →ₗ[K] V₂) :
+  function.injective T.dual_map ↔ function.surjective T :=
+begin
+  refine ⟨_, λ h, dual_map_injective_of_surjective h⟩,
+  rw [← range_eq_top, ← ker_eq_bot],
+  intro h,
+  apply finite_dimensional.eq_top_of_finrank_eq,
+  rw ← finrank_eq_zero at h,
+  rw [← add_zero (finite_dimensional.finrank K T.range), ← h,
+      ← linear_map.finrank_range_dual_map_eq_finrank_range,
+      linear_map.finrank_range_add_finrank_ker, subspace.dual_finrank_eq],
+end
+
+/-- `T` is injective if and only if `T.dual_map` is surjective -/
+lemma injective_iff_dual_surjective
+  [finite_dimensional K V₁]
+  (T : V₁ →ₗ[K] V₂) : function.injective T ↔ function.surjective T.dual_map :=
+begin
+  rw [← range_eq_top, ← ker_eq_bot],
+  refine ⟨λ h, by rw [range_dual_map_eq_dual_annihilator_ker, h];
+               exact submodule.dual_annihilator_bot, _⟩,
```

Comment:
Can you move this `by` block out to a `{ }`d block? Usually we try to avoid `;` unless there are actually multiple goals.